File: src/main/java/dev/alexengrig/leetcode/_128_longest_consecutive_sequence/SetSolution.java
package dev.alexengrig.leetcode._128_longest_consecutive_sequence;

import java.util.HashSet;
import java.util.Set;

class SetSolution implements Task {
    @Override
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }
        int max = 0;
        for (int num : set) {
            if (!set.contains(num - 1)) {
                int length = 1;
                while (set.contains(num + length)) {
                    length++;
                }
                max = Math.max(max, length);
            }
        }
        return max;
    }
}

File: src/main/python/128_longest_consecutive_sequence/set_solution.py
from typing import List

from task import Task


class SetSolution(Task):

    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set(nums)
        res = 0
        for num in seen:
            if num - 1 not in seen:
                l = 1
                while num + l in seen:
                    l += 1
                res = max(res, l)
        return res
